rules.py

```diff
@@ -216,31 +219,84 @@ class ElectoralDistrictOcdId(base.BaseRule):
     _OCDID_URL = (
         "https://raw.github.com/opencivicdata/ocd-division-ids/master/identifiers/country-us.csv"
     )
-
+    github_repo = None
+    
     def __init__(self, election_tree, schema_file):
         super(ElectoralDistrictOcdId, self).__init__(election_tree, schema_file)
+        g = Github()
+        self.github_repo = g.get_repo("opencivicdata/ocd-division-ids")
         self.ocds = self._get_ocd_data()
         self.gpunits = []
         for gpunit in self.election_tree.iterfind("//GpUnit"):
             self.gpunits.append(gpunit)
+            
+    def _get_latest_commit_date(self):
+        """Returns the latest commit date to country-us.csv."""
+        latest_commit_date = None
+        latest_commit = self.github_repo.get_commits(
+            path="/identifiers/country-us.csv")[0]
+        latest_commit_date = latest_commit.commit.committer.date
+        return latest_commit_date
+        
+    def _get_latest_file_blob_sha(self):
+        """Returns the gihub blob sha of country-us.csv."""
+        blob_sha = None
+        dir_contents = self.github_repo.get_dir_contents("identifiers")
+        for content_file in dir_contents:
+            if content_file.name == "country-us.csv":
+                blob_sha = content_file.sha
+                break
+        return blob_sha
+
+    def _download_data(self, file_path):
+        """Makes a request to Github to download the file."""
+        r = requests.get(self._OCDID_URL)
+        with io.open(file_path, "wb") as fd:
+            for chunk in r.iter_content():
```

Comment:
I tried to run it locally and ran into some issues. I'd like to test this 'internally' first before asking NC/partners to do it (if that is okay!)